rpcserver.go

```diff
@@ -1619,6 +1797,13 @@ func (r *rpcServer) ListInvoices(ctx context.Context,
 // notifying the client of newly added/settled invoices.
 func (r *rpcServer) SubscribeInvoices(req *lnrpc.InvoiceSubscription,
 	updateStream lnrpc.Lightning_SubscribeInvoicesServer) error {
+	// Check macaroon to see if this is allowed.
+	if r.server.authSvc != nil {
+		if err := macaroons.ValidateMacaroon(updateStream.Context(),
+			"listinvoices", r.server.authSvc); err != nil {
```

Comment:
See here, for example, where `SubscribeInvoices` actually uses a `listinvoices` permission.